pkg/controller/operators/catalog/operator.go

```diff
@@ -912,41 +917,44 @@ func validateSourceType(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *
 		err = fmt.Errorf("unknown sourcetype: %s", sourceType)
 	}
 	if err != nil {
+		logger.WithError(err).Error("error validating catalog source type")
 		out.SetError(v1alpha1.CatalogSourceSpecInvalidError, err)
 		return
 	}
 
 	// The sourceType is valid, clear all status (other than status conditions array) if there's existing invalid spec reason
 	if out.Status.Reason == v1alpha1.CatalogSourceSpecInvalidError {
+		logger.Info("catalog source has invalid spec")
 		out.Status = v1alpha1.CatalogSourceStatus{
 			Conditions: out.Status.Conditions,
 		}
 	}
 	continueSync = true
-
 	return
 }
 
 func (o *Operator) syncConfigMap(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *v1alpha1.CatalogSource, continueSync bool, syncError error) {
+	logger = logger.WithFields(logrus.Fields{
+		"configmap.namespace": in.Namespace,
+		"configmap.name":      in.Spec.ConfigMap,
+	})
+	logger.Info("checking catsrc configmap state")
+
 	out = in
 	if !(in.Spec.SourceType == v1alpha1.SourceTypeInternal || in.Spec.SourceType == v1alpha1.SourceTypeConfigmap) {
 		continueSync = true
+		logger.Info("catalog type is neither internal not configmap - bailing without error")
 		return
 	}
```

Comment:
I think I'd do the following. Otherwise, it sounds like we'll:
1. Get "checking catsrc configmap state" on every CatalogSource, even if it isn't a configmap type
2. A "catalog type is neither internal..." on all GRPC-based CatalogSources (which are basically all of them)

```suggestion

	out = in
	if !(in.Spec.SourceType == v1alpha1.SourceTypeInternal || in.Spec.SourceType == v1alpha1.SourceTypeConfigmap) {
		continueSync = true
		return
	}
	logger.Info("checking catsrc configmap state")
```

Reply:
Your suggestion is identical to the previous state of the code, by the looks of it.